libsolidity/ExpressionCompiler.cpp

```diff
@@ -200,8 +201,35 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 	else
 	{
 		FunctionType const& function = dynamic_cast<FunctionType const&>(*_functionCall.getExpression().getType());
-		vector<ASTPointer<Expression const>> arguments = _functionCall.getArguments();
-		solAssert(arguments.size() == function.getParameterTypes().size(), "");
+		TypePointers const& parameterTypes = function.getParameterTypes();
+		vector<string> const& parameterNames = function.getParameterNames();
+		vector<ASTPointer<Expression const>> const& callArguments = _functionCall.getArguments();
+		vector<string> const& callArgumentNames = _functionCall.getNames();
+		solAssert(callArguments.size() == parameterTypes.size(), "");
+
+		vector<ASTPointer<Expression const>> arguments;
+		if (callArgumentNames.empty())
+		{
+			// normal arguments
+			arguments = {callArguments.begin(), callArguments.end()};
+		}
+		else
+		{
+			// named arguments
+			for (size_t i = 0; i < parameterNames.size(); i++) {
```

Comment:
The local variable `parameterNames` could be avoided by using a range-based loop here.